server-spi-private/src/main/java/org/keycloak/broker/provider/util/SimpleHttp.java

```diff
@@ -273,6 +274,8 @@ private void readResponse() throws IOException {
                 HttpEntity entity = response.getEntity();
                 if (entity != null) {
                     is = entity.getContent();
+                    MediaType mediaType = MediaType.valueOf(entity.getContentType().getValue());
```

Comment:
Since this class doesn't rely on JAX-RS I'd prefer it to use httpclient approach. See here https://stackoverflow.com/questions/9112259/obtaining-response-charset-of-response-to-get-or-post-request/34281952#34281952.

We also need to add some test to make sure this actually fixes the issue and that there is no regression in the future. It'd be hard to write a proper integration test around this, so a unit test for SimpleHttp would be sufficient.

Reply:
> Since this class doesn't rely on JAX-RS I'd prefer it to use httpclient approach. See here https://stackoverflow.com/questions/9112259/obtaining-response-charset-of-response-to-get-or-post-request/34281952#34281952.

Thank. Looks pretty.

> We also need to add some test to make sure this actually fixes the issue and that there is no regression in the future. It'd be hard to write a proper integration test around this, so a unit test for SimpleHttp would be sufficient.

Done.